rj_param_utils/testing/src/gpp_test.cpp

```diff
@@ -0,0 +1,156 @@
+                 kExampleStringDescription2)
+DEFINE_NS_FLOAT64_VEC(kGlobalNode, test::byte, namespaced_double_vec, kExampleDoubleVecValue,
+                      kExampleDoubleVecDescription)
+
+class ReceiverNode : public rclcpp::Node {
+public:
+    ReceiverNode() : rclcpp::Node("receiver_node") {};
+
+    ::params::ROS2GlobalParamProvider global_param_provider_ = ::params::ROS2GlobalParamProvider(this, kGlobalNode);
+};
+
+class GlobalParamProviderTest : public ::testing::Test {
+public:
+    void SetUp() override {
+        rclcpp::init(0, {});
+
+        global_node_ = std::make_shared<rclcpp::Node>(kGlobalNode);
+        receiver_node_ = std::make_shared<ReceiverNode>();
+        global_node_->declare_parameter("global_bool", true);
+        global_node_->declare_parameter("global_float", 3.14f);
+
+        executor_ = std::make_shared<rclcpp::executors::SingleThreadedExecutor>();
+
+        executor_->add_node(global_node_);
+        executor_->add_node(receiver_node_);
+    }
+
+    void TearDown() override {
+        is_running_ = false;
+        ros_thread_.join();
+        rclcpp::shutdown();
+    }
+protected:
+    void start_ros() {
+        is_running_ = true;
+        ros_thread_ = std::thread{[this]() {
+            while (is_running_ && rclcpp::ok()) {
+                executor_->spin_once();
+            }
+        }};
+    }
+
+    rclcpp::executors::SingleThreadedExecutor::SharedPtr executor_;
+    std::shared_ptr<rclcpp::Node> global_node_;
+    std::shared_ptr<ReceiverNode> receiver_node_;
+    std::thread ros_thread_;
+    std::atomic<bool> is_running_;
+};
+
+/**
+ * @brief Test that the default value of DEFINE_* variant of defining
+ * params is correct.
+ */
+TEST_F(GlobalParamProviderTest, RequestUpdateParam) {
+    start_ros();
+    ASSERT_TRUE(PARAM_bare_bool);
+    ASSERT_EQ(PARAM_bare_double, kExampleDoubleValue);
+}
+
+TEST_F(GlobalParamProviderTest, CorrectBareDefaultBareValue) {
+    start_ros();
+    EXPECT_EQ(PARAM_bare_bool, kExampleBoolValue);
+    EXPECT_EQ(PARAM_bare_int64, kExampleIntValue);
+    EXPECT_EQ(PARAM_bare_double, kExampleDoubleValue);
+
+    ASSERT_EQ(PARAM_bare_byte_vec.size(), kExampleByteVecValue.size());
+    for (size_t i = 0; i < PARAM_bare_byte_vec.size(); i++) {
+        EXPECT_EQ(PARAM_bare_byte_vec[i], kExampleByteVecValue[i]);
+    }
+
+    ASSERT_EQ(PARAM_bare_bool_vec.size(), kExampleBoolVecValue.size());
+    for (size_t i = 0; i < PARAM_bare_bool_vec.size(); i++) {
+        EXPECT_EQ(PARAM_bare_bool_vec[i], kExampleBoolVecValue[i]);
+    }
+
+    ASSERT_EQ(PARAM_bare_int64_vec.size(), kExampleIntVecValue.size());
+    for (size_t i = 0; i < PARAM_bare_int64_vec.size(); i++) {
+        EXPECT_EQ(PARAM_bare_int64_vec[i], kExampleIntVecValue[i]);
+    }
+
+    ASSERT_EQ(PARAM_bare_string_vec.size(), kExampleStringVecValue.size());
+    for (size_t i = 0; i < PARAM_bare_string_vec.size(); i++) {
+        EXPECT_EQ(PARAM_bare_string_vec[i], kExampleStringVecValue[i]);
+    }
```

Comment:
You need something in these tests that does something like:
```cpp
global_node_->set_parameter(rclcpp::Parameter("bare_bool", false);
```
and then asserts that the proper value is changed.